backend/getDisplayName.js

```diff
@@ -18,7 +18,23 @@ function getDisplayName(type: Function): string {
     return cachedDisplayNames.get(type);
   }
 
-  let displayName = type.displayName || type.name || 'Unknown';
+  let displayName = type.displayName || type.name;
+
+  // The displayName property might not be a string.
+  // If it's a custom function, attempt to use its return value.
+  // github.com/facebook/react-devtools/issues/803
+  if (typeof displayName === 'function') {
+    try {
+      displayName = displayName();
```

Comment:
Hmm, do we truly support this? I don't think we ever did. I think it's fine to just ignore it in this case—not sure we'd want to call it. For example we don't in the error messages.